app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/uploadsupportingdocs/UploadSupportingEvidenceController.scala

```diff
@@ -0,0 +1,1012 @@
+                logger.warn("Could not update session", e)
+                errorHandler.errorResult()
+              }, _ => Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers()))
+
+            } else {
+              val result = for {
+                _ <- upscanService.removeAllFiles(DraftReturnId(draftReturnId.toString))
+                updatedAnswers: UploadSupportingEvidenceAnswers = answers match {
+                  case IncompleteUploadSupportingEvidenceAnswers(_, _) =>
+                    IncompleteUploadSupportingEvidenceAnswers(Some(false), List.empty)
+                  case CompleteUploadSupportingEvidenceAnswers(_, _) =>
+                    CompleteUploadSupportingEvidenceAnswers(doYouWantToUploadSupportingEvidence = false, List.empty)
+                }
+                newDraftReturn = fillingOutReturn.draftReturn match {
+                  case s @ DraftSingleDisposalReturn(
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _
+                      ) =>
+                    s.copy(uploadSupportingDocuments = Some(updatedAnswers))
+                  case m @ DraftMultipleDisposalsReturn(
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _,
+                        _
+                      ) =>
+                    m.copy(uploadSupportingDocuments = Some(updatedAnswers))
+                }
+
+                _ <- returnsService
+                      .storeDraftReturn(
+                        newDraftReturn,
+                        fillingOutReturn.subscribedDetails.cgtReference,
+                        fillingOutReturn.agentReferenceNumber
+                      )
+                _ <- EitherT(
+                      updateSession(sessionStore, request)(
+                        _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = newDraftReturn)))
+                      )
+                    )
+              } yield ()
+
+              result.fold({ e =>
+                logger.warn("Could not update session", e)
+                errorHandler.errorResult()
+              }, _ => Redirect(routes.UploadSupportingEvidenceController.checkYourAnswers()))
+
+            }
+        )
+      }
+  }
+
+  def uploadSupportingEvidence(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    withUploadSupportingEvidenceAnswers(request) { (draftReturnId, cgtRef, _, _, _) =>
+      upscanService.initiate(DraftReturnId(draftReturnId.toString), cgtRef, LocalDateTime.now()).value.map {
+        case Left(error) =>
+          error.value match {
+            case Left(_) => errorHandler.errorResult(None)
```

Comment:
you shouldn't need to supply `None` to `errorResult`. You should be able to just to `errorHandler.errorResult()` and it should find the user type it wants implicitly

Reply:
looks like you have taken these out already :+1:

Reply:
yup, took them out